app/Http/Controllers/SurveyController.php

```diff
@@ -716,21 +718,38 @@ public function storeResult(ResultRequest $request)
         DB::beginTransaction();
 
         try {
+            $survey = $this->surveyRepository->getSurveyFromToken($request->json()->get('survey_token'));
+
+            if (Carbon::now()->gt(Carbon::parse($survey->end_time))
+                || Carbon::now()->lt(Carbon::parse($survey->start_time))) {
+                $redirect = route('survey.create.complete', $survey->token);
+                throw new Exception('Do not permit to do this survey');
+            }
+
             $this->resultRepository->storeResult($request->json(), $this->surveyRepository);
             $request->session()->forget('current_section_survey');
 
             DB::commit();
 
-            return response()->json([
+            $dataJSON = [
                 'success' => true,
-            ]);
+            ];
         } catch (Exception $e) {
             DB::rollback();
 
-            return response()->json([
-                'success' => false,
-                'message' => trans('lang.send_result_failed'),
-            ]);
+            if (isset($redirect)) {
```

Comment:
đoạn này có thể viết như này thì clean hơn nè

```suggestion
            $dataJson['success'] = true;
            
            if (!isset($redirect)) {
                $dataJson['success'] = false;
                $dataJson['message'] = trans('lang.send_result_failed');
            }
```